### DISCUSSION TITLE
tool for architecture design?

### DISCUSSION THREAD

**Discussion by devzohaib** [neuroneural/brainchop#14] [General]
Which tool do you use to create model architecture design?

**Comment by Mmasoud1** [comment]
We have a data flow architecture combining functional and object oriented programming.  The UI created with Webix

**Comment by niyaziacer** [comment]
Dear Mohamed 
Thank you for Brainchop. I like it. I used your data as t1, everything was fine. because your data size 256x256x256 and 1 mm slice thickness. 
but I have different size Mr data. I uploaded your sisytem but it failed. What can I do it? if I use my Mr data. what is your suggestion?
many thanks for your answer. 

Niyazi Acer

**Reply by niyaziacer** [reply]
I read your paper mri_convert.Js. But I have no idea how can I work. many thanks

**Reply by Mmasoud1** [reply]
Thank you Niyazi for your feedback. We are really glad you like Brainchop and proud to have you as a member of Brainchop family. 

The tool is designed  to mainly depend on the end-user H/W and S/W in use. In general, external GPU is highly recommended especially for models  "Full Brain GWM (large)",  "FS aparc+aseg Atlas 104 (failsafe)", "Cortical Atlas 50".  For example the GPU  "GeForce GTX 1050 Ti"  is  able to run all the models.  

Therefore, my first suggestion  is to try a machine with an external good GPU if possible. You can look also in this [list](https://docs.google.com/spreadsheets/d/1PU4p6oN4aBfTbXZ8pO6n8fc890Y5Ck0agsIZbm_FO3I/edit#gid=0) of verified H/W and S/W for each model. 

If you are using an internal or built-in graphics chip such as  HD or UHD,  they can run mostly  models such as:  "Full Brain GWM (light)", "Compute Brain Mask (FAST)",  "Extract the Brain (FAST)".  If our sample can run with your machine for any of  these models, then any other sample should run also. 

Sometimes a  built-in  graphics chip (e.g. HD,  UHD, etc )  can run also   "Cortical Atlas 50" with our sample,  but changing the sample with your sample  in that model "Cortical Atlas 50" may not run sometimes. The reason for that is  because in that model particularly  we use a cropping preprocessing stage which means  we are trying to exclude all the black empty volume around the brain data before feeding the cropped brain to the model for inference. If that empty volume is large enough such as the case with our sample then the model may run safely,  but if the empty volume is small and the brain volume is large with a different sample by user then the model may fail due to the limited  built-in  graphics chip resources.  If you have a  built-in graphics chip and our sample run with you for "Cortical Atlas 50"  model, then your samples need to be tested one by one. 

I highly recommend trying an external GPU if possible with at least 16 GB RAM ( see [System Requirements ](https://github.com/neuroneural/brainchop/wiki/System-Requirements) ) . Otherwise you will need to take care of above comments while trying built-in graphics chips. 

I will be happy to follow up with you for full Brainchop utilization. Let me know please if you have any questions or need any further clarification.

**Reply by Mmasoud1** [reply]
For mri_convert.Js,  it runs automtically with Brainchop  if there is a need for that.  Once your sample needs such a conversion you will see a trigger window telling you a conversion is needed with your sample and the conversion will run with your "OK" click. 

Please check this post and associated video for more info: 
https://www.linkedin.com/feed/update/urn:li:activity:6943609801499193345/?utm_source=share&utm_medium=member_desktop

**Comment by niyaziacer** [comment]
Thank you I did
Firstly I selected full brain GWM. I saved converted file as 256*256 size.
But Memory is big problem. I have 16 gb RAM. I will try again. I will got
it.
Thank you


On 28 Jan 2023 Sat at 21:02 Mohamed E. Masoud ***@***.***>
wrote:

> For mri_convert.Js, it runs automtically with Brainchop if there is a need
> for that. Once your sample needs such a conversion you will see a trigger
> window telling you a conversion is needed with your sample and the
> conversion will run with your "OK" click.
>
> Please check this post and associated video for more info:
>
> https://www.linkedin.com/feed/update/urn:li:activity:6943609801499193345/?utm_source=share&utm_medium=member_desktop
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/neuroneural/brainchop/discussions/14#discussioncomment-4805232>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIXHJVM43ATO5NXBJPLGKODWUVNKDANCNFSM6AAAAAAQUGPZLI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>
-- 
Prof.Dr. Niyazi ACER
Arel Üniversitesi
Tip Fakültesi
Anatomi Anabilim Dalı
İstanbul, Türkiye

Department of Anatomy
Faculty of Medicine
Arel University
Istanbul- TURKEY
Telephone: +90 533 389 96 76
E mail: ***@***.*** ***@***.***>
          ***@***.*** ***@***.***>

**Reply by Mmasoud1** [reply]
Make sure  please what type of graphics card you are using, is it built-in graphics chip (HD, UHD)  or external one (e.g. Nvidia GeForce GTX xxxx  or similar) ?   If you don't have external GPU then  it is hard to run relatively large models such as  "Full Brain GWM (large)", "FS aparc+aseg Atlas 104 (failsafe)" or  "Cortical Atlas 50".

**Reply by niyaziacer** [reply]
I use Nvidia geforce RTX. It start process after 5 minutes it says "context lost due to limited memory avaliable". I dont undertsant but I will try again.

**Reply by Mmasoud1** [reply]
Please try this:  

Open Brainchop in the browser again --->  press F12 to see the browser console --->  In the browser console run this command : 

detectGPUCardType()

Does the command output show  Nvidia geforce RTX  or it just detects your build-in graphics chip (e.g. HD or UHD) ?

**Reply by Mmasoud1** [reply]
Sometimes we may have external GPU but the browsers can only  detect and use  the built-in graphics card, to fix that, we need first to check which GPU type is in use by the browser.  We can check that by any following methods:

-  Enter **about:support** in the browser URL bar and check the Graphics section.

Or

- Open Brainchop ---> press F12 to see the browser console ---> In the browser console run this command :

detectGPUCardType()

This should tell us if the browser is properly use our external GPU  or no.  If not then we can go to the next step to fix the problem

**Comment by niyaziacer** [comment]
I use firefox and solved problem. many thanks
good nights


Mohamed E. Masoud ***@***.***>, 28 Oca 2023 Cmt, 23:03
tarihinde şunu yazdı:

> Sometimes we may have external GPU but the browsers can only detect and
> use the built-in graphics card, to fix that, we need first to check which
> GPU type is in use by the browser. We can check that by any following
> methods:
>
>    - Enter *about:support* in the browser URL bar and check the Graphics
>    section.
>
> Or
>
>    - Open Brainchop ---> press F12 to see the browser console ---> In the
>    browser console run this command :
>
> detectGPUCardType()
>
> This should tell us if the browser is properly use our external GPU or no.
> If not then we can go to the next step to fix the problem
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/neuroneural/brainchop/discussions/14#discussioncomment-4805710>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AIXHJVIWLIJ7WBNQULRWXLDWUV3QZANCNFSM6AAAAAAQUGPZLI>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
Prof.Dr. Niyazi ACER
Arel Üniversitesi
Tip Fakültesi
Anatomi Anabilim Dalı
İstanbul, Türkiye

Department of Anatomy
Faculty of Medicine
Arel University
Istanbul- TURKEY
Telephone: +90 533 389 96 76
E mail: ***@***.*** ***@***.***>
          ***@***.*** ***@***.***>

**Reply by Mmasoud1** [reply]
You are welcome. My email mmasoud2@outlook.com, please feel free to contact me any time and we can setup zoom meeting for quick troubleshooting any Brainchop issue you may face in the future.   Good night :)
